### PR TITLE
Tests for max-stale request cache-directive

### PR DESCRIPTION
**Pull request by stanculescu**
Relevant implementation PR: https://github.com/reaktivity/nukleus-http-cache.java/pull/28

### CODE REVIEW

File: src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/cache.max-stale.no.value/accept/client.rpt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        "nukleus://http-cache/streams/source"
+        option nukleus:route ${newProxyAcceptRef}
+        option nukleus:window 8192
+        option nukleus:transmission "half-duplex"
+
+write nukleus:begin.ext ${http:header(":scheme", "http")}
+                        ${http:header(":method", "GET")}
+                        ${http:header(":path", "/path")}
+                        ${http:header(":authority", "localhost:8080")}
+                        ${http:header("cache-control", "max-age=1, max-stale")}
```
Comment (dpwspoon) [inline src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/cache.max-stale.no.value/accept/client.rpt]: why does this have the max-age directive?  Does the spec specify what we should do if there is both of these and which one takes precident?

Comment (dpwspoon) [reply src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/cache.max-stale.no.value/accept/client.rpt]: Section 5.2.1.1.
```
5.2.1.1.  max-age

   Argument syntax:

      delta-seconds (see Section 1.2.1)

   The "max-age" request directive indicates that the client is
   unwilling to accept a response whose age is greater than the
   specified number of seconds.  Unless the max-stale request directive
   is also present, the client is not willing to accept a stale
   response.

   This directive uses the token form of the argument syntax: e.g.,
   'max-age=5' not 'max-age="5"'.  A sender SHOULD NOT generate the
   quoted-string form.
```

But it seems like you already have a test for that below, so probably no need for that directive here.

File: src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/cache.max-stale/accept/client.rpt [hunk 2]
```diff
@@ -0,0 +1,73 @@
+#
```
Comment (dpwspoon) [inline src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/cache.max-stale/accept/client.rpt]: nit, consider renaming this script cache.max-stale.with.value to be inline with the without value script

File: src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/expire.max-stale/accept/client.rpt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                       ${http:header("last-modified", date)}
+
+read "hello world"
+
+read closed
+read notify REQUEST_CACHED
+
+
+# INPUT STREAM 2
+connect await CACHE_EXPIRED
```
Comment (dpwspoon) [inline src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/expire.max-stale/accept/client.rpt]: nit: consider renaming property CACHE_EXPIRED_AND_STALE_FOR_2_SECONDS or something like that??

**Review by dpwspoon — CHANGES_REQUESTED** [review]
Very minor feedback.  LGTM.  Thanks

**Comment by dpwspoon** [comment]
Very thorough set of test cases.  Thanks @stanculescu

**Review by dpwspoon — DISMISSED** [review]

File: src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/cache.max-stale.no.value/accept/client.rpt [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+write close
+
+read nukleus:begin.ext ${http:header(":status", "200")}
+                       ${http:header("content-length", "11")}
+                       ${http:header("cache-control", "max-age=15")}
+                       ${http:header("date", date)}
+                       ${http:header("last-modified", date)}
+                       ${http:header("warning", "110 - \"Response is Stale\"")}
+
```
Comment (akrambek) [inline src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/cache.max-stale.no.value/accept/client.rpt:70]: If the max-age is 15 seconds and not yet expired I wonder it should receive `Response is Stale` response. By looking into implementation code test `shouldCacheMaxStaleWithNoValue` from `nukleus-http-cache.java` it looks like you meant `max-age=1` and if so why is this passing? Would you please clarify this?

Comment (akrambek) [reply src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/cache.max-stale.no.value/accept/client.rpt:70]: I think it makes sense to add another Input Stream to test if it is working for any age as specified in the rfc which makes test case complete.

**Pull request merged**